### PR TITLE
Upgrade google-cloud-nio to fix broken requester pays support

### PR DESCRIPTION
**Pull request by lbergelson**
### Description

This upgrade the google-cloud-nio version to match the one that GATK is using (well, the one gatk is moving too).  It also switches from using the shaded version -> standard version.  We initially used the shaded version because of horrible downstream dependency issues in GATK, but those have been resolved so we can switch to the unshaded version now.  

When using the shaded version we ignored transitive dependencies since the actual dependencies were included in the shaded jar and the transitive ones imported additional unnecessary things.  Switching to unshaded means we need the transitive ones.  

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

**Comment by droazen** [comment]
Would be great to get a quick review/merge/release of this PR from a Picard maintainer (is that still you @gbggrant ?) Due to an internal change Google made recently, Google Cloud Storage requester pays support is broken in Picard in the ancient version of the Google NIO library it's on. This PR should fix that issue, and also fix integration with GATK, which has moved to the newer release of google-cloud-nio.

**Review by gbggrant — APPROVED** [review]

**Comment by droazen** [comment]
Thanks for the quick turnaround @gbggrant ! Is it possible to get a Picard release with this change?

**Comment by gbggrant** [comment]
Yeah, working on it. I've had to regenerate my signing keys, so hoping I won't run into any issues.

**Comment by ldgauthier** [comment]
@gbggrant can we hold off on the release until tomorrow?  I was to get #1786 in, which should be one last quick review.

**Comment by droazen** [comment]
@gbggrant I just received a request from @ldgauthier to hold off on the Picard release until one of her PRs can be merged -- could we delay by 1 day? Sorry for the confusion!

**Pull request merged**
